pages/static/serviceworker.js

```diff
@@ -9,3 +10,67 @@ AMP_SW.init({
     assets: [],
   },
 });
+
+const CACHE_NAME = 'AMP-DEV-SEARCH-CACHE';
+const LATEST_QUERY_URL = '/search/latest-query';
+
+
+// Match search request url
+const matchDo = ({url}) => {
+  return (url.pathname === '/search/do');
+};
+
+// Handle search request
+const handlerDo = async ({url, event}) => {
+  // Extract query parameter from search request url
+  // Put new query as response to AMP-DEV-SEARCH-CACHE
+  const searchQuery = url.search.match(/q=([^&]+)/)[1];
+  const cache = await caches.open(CACHE_NAME);
+
+  cache.put(LATEST_QUERY_URL, new Response(`"${searchQuery}"`));
+
+  let response = await cache.match(event.request);
+
+  // If search results are already in cache, skip
+  // Else fetch new search request and cache whole results
+  if (response) {
+    return response;
+  } else {
```

Comment:
nit: `else` is no needed here => less indentation

Reply:
Thanks, learned something new 👍